docs/user-guide/src/bmo/ironic_installation.md

```diff
@@ -0,0 +1,181 @@
+# Install Ironic
+
+Baremetal Operator must be able to interact with Ironic. First
+we will see what are the Ironic optional containers we need to
+run. After we will go through Ironic installation in-cluster and
+out-of-cluster. In any case, there are seven containers that need
+to run to be able to provision baremetal nodes.
+
+- ironic-dnsmasq
+- ironic-api
+- ironic-conductor
+- ironic-log-watch
+- ironic-inspector
+- ironic-ipa-downloader
+- mariadb
+
+To know more about each container's functionality check the documentation
+[here](https://github.com/metal3-io/ironic-image#description).
+
+## Prerequisites
+
+Container runtime (e.g., docker, podman). Below we use docker.
+
+<!-- include these inline so we can style an match variables -->
+{{#include ironic_variables.md}}
+
+## Optional containers
+
+In some circumstances, it might be necessary to install some additional containers on
+top of Ironic containers. And this might depend on whether your target nodes are
+virtual or physical machines, or whether you want to use more stable BMC protocol
+like Redfish and etc. Let's check why we would need these optional containers.
+We run these optional containers on the host machine, i.e., out-of-cluster regardless
+of the Ironic installation method.
+
+### vbmc
+
+If you plan to manage [livbirt](https://en.wikipedia.org/wiki/Libvirt) Virtual Machine (VM),
+there is a high chance that you need to install Virtual Baseboard Management
+Controller (VBMC), which is virtual implementation of Baseboard Management
+Controller (BMC). For More info on VBMC check official [documentation](https://docs.openstack.org/virtualbmc/latest/).
```

Comment:
Add a reference link to BMC as well.

Reply:
Would this be good: https://www.ibm.com/docs/en/power9?topic=interfaces-managing-bmc-based-systems

Reply:
The link I provided seems to be quite specific and I could not get a good reference on the matter. 
Would you please ignore my comment and keep that part as is.
Thanks.